0010-zkapp-cli-lightnet-management.md

```diff
@@ -0,0 +1,100 @@
+- Reduced complexity and better DX, eliminating manual 3rd party tools management.
+
+## Detailed design
+
+### zkapp CLI Interface Enhancements
+
+Under the primary command `zk`, the new `lightnet` sub-commands will be introduced:
+
+#### Help
+
+- **Command**: `zk lightnet`
+- **Output**: Help message with description of available sub-commands and their options.
+
+#### Start lightnet network
+
+- **Command**: `zk lightnet start <options>`
+- **Function**: Checks the env., starts the local lightnet Docker container with reasonable default configuration options and waits until the network sync. Repeating the command while network is up and running should not start a new container, but rather check the status of the existing one.
+- **Configuration options**:
+  - --mode: `single-node` or `multi-node` (default: `single-node`)
+  - --proof-level: `none` or `full` (default: `none`)
+  - --archive-node: `true` or `false` (default: `true`)
+  - --mina-branch: `rampup`, `berkeley` or `develop` (default: `rampup`)
+  - --dune-profile: `lightnet` or `devnet` (default: `lightnet`)
+- **Output**:
+  - Anticipated network properties (as per [Docker image description](https://hub.docker.com/r/o1labs/mina-local-network));
+  - Mina Daemon GraphQL endpoint for communication: [http://localhost:8080/graphql](http://localhost:8080/graphql)
+  - PostgreSQL connection string: `postgresql://postgres:postgres@localhost:5432/archive`;
+  - Mina Daemon and Archive Node logs path;
+  - [o1js configuration](https://github.com/o1-labs/o1js/blob/ccf50e0b58190d9700c6dda4fafee3e62e270131/src/examples/zkapps/hello_world/run_live.ts#L18) code snippet.
+
+#### Stop lightnet network
+
+- **Command**: `zk lightnet stop`
+  - In future we might want to extend this command with additional options to preserve the network state between containers lifecycle.
+- **Function**: Destroys the running container. Repeating the command while network is down should not throw an error but rather output a message that the network is already down.
+- **Output**: Maybe blockchain state (blocks height, slots, epoch, txns processed during the session, etc.) at the time of command issuing.
+
+#### Handle logs
+
+- **Command**: `zk lightnet logs <sub-command>`
+- **Sub-commands**:
+  - **view**: `less` or `tail` the log files (Mina Daemon or Archive Node). OS agnostic.
+  - **save**: Saves the log files to default location and outputs the resulting path.
+
+#### Manage accounts
+
+- **Command**: `zk lightnet account <sub-command>`
+- **Sub-commands**:
+  - **fetch**: Fetches available account from the container's pool.
+  - **release**: Releases "used" account back to the pool.
+- **Note**: Corresponding [o1js API](https://github.com/o1-labs/o1js/pull/1167) can be used.
+
+#### Launch the blockchain explorer
+
+- **Command**: `zk lightnet explorer`
```

Comment:
@shimkiv just confirming that this functionality will be added once everything else here has been released, and after archive node API. Correct?

Reply:
Correct.